### ISSUE TITLE
rtl8821ae constantly dropping connection

### ISSUE THREAD

**Issue by pgueckel** [lwfinger/rtlwifi_new#235]
I filed a bugzilla report at Fedora:

https://bugzilla.redhat.com/show_bug.cgi?id=1435793

but there has been no action. I was finally told to contact upstream if I wanted this fixed.

This morning I upgraded to kernel 4.11.0-0.rc8.git0.1.fc26.x86_64, but it has not changed—the problem persists.

**Comment by pgueckel** [comment]
There is a suggestion that fedora bug 1382741 (https://bugzilla.redhat.com/show_bug.cgi?id=1382741) is the same as the one I reported above. There has been some discussion, so maybe someone will be able to help.

**Comment by pgueckel** [comment]
It is now thought that fedora bug 1382741 is not the same as this, but this is exactly the same bug as fedora bug 1435793 (since I reported it!).

**Comment by wyldcard** [comment]
I have the same problem on my ubuntu 16.04 machine. Its almost unusable. I'm also forced to use 2.4ghz because it simply wont connect to 5ghz even tho its visible. Im using the ASUS PCE-AC51 card.

**Comment by lwfinger** [comment]
You do realize that this kind of complaint with no documentation does no good. Which card do you have? RTL8821AE or RTL8812AE? What country code does the card report?

On my RTL8821AE, I can connect to either of my 5G APs. The connection does drop, but that is nothing I can fix. The internal workings of the chip are unknown to me, thus only Realtek can fix that part.

I do not have an RTL8812AE.

**Comment by wyldcard** [comment]
Im sorry I realized that I wrote in haste with no feedback. I pulled your drivers I think 4 days ago, and I did not see your update to the RTL8821AE 2 days ago. Google led me straight when I were researching and I posted a quick heads up comment.

I tried your updated driver and I can now connect to the 5Ghz! Your comment popped up as I were typing to let you know your update works. Sorry for the crap comment, you are doing a fantastic job!

**Comment by pgueckel** [comment]
KDE KInfocenter reports that I have the RTL8821AE 802.11ac PCIe wireless adapter. My computer is an Asus F555LA laptop. I bought it new in Canada in August 2016. I have no idea how to determine the country code or how to get it to "report".

I am running Fedora 26 alpha, with all of the updates/test updates current to this minute. The current kernel is 4.11.0-0.rc8.git0.1.fc26.x86_64. Further network-related: wpa_supplicant-2.6-6.fc26.x86_64, NetworkManager-1.8.0-0.2.rc3.fc26.x86_64.

I have disabled 2.4G on my router and use only 5G (both work, both have the same problem). I just don't want to unnecessarily clog the airwaves in my building with signals I never use. I only have 3 devices that connect wirelessly: a Panasonic Blu-ray player, a Moto G4 Plus smart phone, and this Asus F555LA laptop. They can all use 802.11n. Only the laptop can use 802.11ac, so I decided to select 802.11n for the broadcasting (I read that, despite backward compatibility, it is better that all devices use the same protocol. Is it really necessary?).

Anyway, the phone and the blu-ray player have absolutely no problem connecting and maintaining a connection at 5G (and, formerly, at 2.4G). The laptop establishes a connection and it is sort of steady for a bit, a few minutes, a few web pages, sometimes barely a couple of seconds, and then NetworkManager suddenly displays no strength bars at all and the network is blocked (for the laptop, only!). It takes about a minute for NetworkManager then to display a message stating that the network is unavailable and that it has been disconnected. Then, it just meditates for another minute or so (seems like a ridiculously long time, on both counts!) and then it attempts to reconnect, succeeds instantly and the same symptoms start over again.

I don't know who is spreading this dyslexic info about RTL8812AE, because this is not! It is RTL8821AE.

**Comment by pgueckel** [comment]
I am very interested in this new driver you have and have no idea how to get it or how to install it. Is it in the Fedora repository?

**Comment by pgueckel** [comment]
Oh, and I never had any problems at all with this card (no network loss, etc) until the last of the 4.9 kernels. I don't recall if it was the last 4.9 that still worked or the second-last 4.9. At first, I guess it just started dropping the connection and I didn't think too much of it, except that "this is new and never happened before", but it kept reconnecting, so I brushed it off as not noteworthy. I should have kept more accurate track :-( Kernels were issued every day or two or three from the fedora updates-testing repo (I was still using Fedora 25 then, back in March) and I upgraded as they came out. In any case, from kernel 4.10 onward, this problem has been persistent and unchanging. It was then that I realized that this was a problem that wasn't going away. I panicked and disabled the fedora updates-testing repo altogether and did a distro-sync to get my system back to the official updates (no more testing!). It didn't help, since the straight repo was pretty much caught up, with a lag of a day or three, to the updates-testing repo. I decided to post a bug report.

Pretty much the day after posting the report, I learned that Fedora 26 alpha had been released. I always keep a spare partition for testing, so I overwrote the old Fedora 24 with Fedora 26 alpha, keeping my working Fedora 25 system intact. Well, since swap and efi got reformatted, I was unable to get back into Fedora 25 (lol I got it working again a couple of days later, but there is no longer any point in me using an old release), so I have been using Fedora 26 ever since.

So, the problem began with the last of the Fedora 4.9 kernels in Fedora 25 and it was present in Fedora 26 from the get go.

**Comment by wyldcard** [comment]
The driver is here on lwfinger's gitgub repo.
A quick rundown how I did it...

git clone http://github.com/lwfinger/rtlwifi_new.git
cd rtlwifi_new
make
sudo make install

Load the module with:
modprobe rtl8821ae

You might want to add the module for loading upon boot, and if the old driver keeps loading blacklist it

**Comment by lwfinger** [comment]
@pgueckel: If you want to help debug the problem, you need to obtain the git repo for the Fedora kernels and bisect the problem. If that is beyond your technical skills then you need to tell me exactly which Fedora kernel started the failure. No, 4.9 is not sufficient.

@wyldcard: Once you do the "sudo make install" step, the original (kernel) driver is overwritten. Adding a blacklist at that point will just cause no driver to load.

**Comment by pgueckel** [comment]
@lwfinger I would like to help debug the problem. I have the computer with the affected hardware. I don't have a clue how to "obtain the git repo for the Fedora kernels" and I don't have a clue what it is to "bisect" the problem.

I can also not tell you *exactly* which Fedora (25) kernel started the failure. I am just looking through koji and I can see that the last two 4.9 kernels for Fedora 25 were kernel-4.9.13-201.fc25 and kernel-4.9.14-200.fc25. The first 4.10 kernel for Fedora 25 was kernel-4.10.4-200.fc25. It pretty much has to be one of these. I know that I was onto the problem pretty quickly, even if I did dally (convince myself that there was nothing wrong :-( ) for a few days.

**Comment by pgueckel** [comment]
Correction: the first 4.10 kernel was kernel-4.10.3-200.fc25. So, the closest I can get is the last two 4.9s listed above and this one.

**Comment by lwfinger** [comment]
You should be able to reinstall the 4.9.13 and 4.9.14 kernels from koji and test each of them to see if that is where the problem occurs. As to obtaining the git repo for Fedora, that is a question for Fedora. I do not use that repo.

**Comment by pgueckel** [comment]
@lwfinger I did as you suggested—in Fedora 25!!! That is where the problem first occurred and it has never NOT OCCURRED in Fedora 26.

I installed:

kernel-4.9.13-200.fc25.x86_64 (works just fine, tested over an hour)
kernel-4.9.13-201.fc25.x86_64 (seems to work fine, but only tested for about 5-10 minutes)
kernel-4.9.14-200.fc25.x86_64 (HAS THE BUG)
kernel-4.10.3-200.fc25.x86_64 (not tried, but all subsequent kernels were tainted during the period I used Fedora 25, prior to migrating to Fedora 26 alpha)

This problem has also been manifest on ALL Fedora 26 kernels since the release of Fedora 26 alpha. What this means is that ALL Fedora kernels on both Fedora 25 and Fedora 26 since early March 2017 are BAD!

I hope this narrows it down enough for you to establish the error. If you wish, I can test 4.9.13-201 a little more, but, as I indicated, it seemed to be fine during the few minutes I did try it.

I do not use Fedora 25 any more, but this problem affects Fedora 26 from the get go (from the release of alpha, anyway), so this is most dire.

**Comment by ferviron** [comment]
After searching the internet for a couple of days I finally found this issue, which seems to be quite widespread. Here goes my case: I have an ASUS laptop (bought in early December) which requires the rtl8821ae driver. I used to dual boot Windows and Archlinux. Everything was completely fine up until early March. In late February I started to triple boot Windows, Arch and Void Linux. 

A week after I installed Void, my wifi started to casually drop after a few hours of usage. I though back then, that this was an issue unique to Void, since it is not very stable nor tested. However, I started getting the same issue with Archlinux (though less often), then is when I realized, that either a firmware update or a kernel update could have caused the problem. Regarding Windows: it gave me quite a few issues, but all were easily solved by unloading and reactivating the driver, after a couple of incidents with it, no more problems occurred. 

Fast forward to right now. I have kept up with the issue for quite a while now, but no any longer. In the past few weeks these random disconnects are happening quite more often. What used to take some hours to occur, now the even happen a couple of minutes after boot, in Void and in Arch.

Here goes some _debugging_ information: 

* What happens: wifi drops after some time and does not come back, no matter what. It does boot normally, this issue happens after some time has passed. I can correctly connect to 5GHz wifi with no issue (until it fails). Download speeds are beautifully high.

* Tools installed: wpa_supplicant, network manager, rfkill, the rtl8821ae driver, 4.10 kernel (in both systems) and 4.4LTS in Void too (which also fails!).

* What have I tried after it fails: turn off turn on in network manager, nothing. Unloading and reloading the module, nothing, more info later about the error dmesg spits. Reboot, it works, until it fails. 

* What I believe causes this issue: either a kernel update (not likely since 4.4 also exhibits issues), a firmware update, or simple instability in the driver. I must add that my wifi connection is not he most stable one, though it never gave me issues.

Some relevant output: dmesg: I currently do not have the exact code, but here are some pointers. It disconnects and logs a message telling that the connection with the host was lost, and then it starts complaining about the init-MAC, which  keeps spaming continuously, until the computer shuts down.

ip link shows no state (no up nor down) after the failure.

Network manager displays a wifi not ready or disconnected after the connection drops.

dmesg after reloading the module: It complaints that the power state is 3D and cannot be changed (I believe it has to go to D0), then the MAC message goes on. I'll report back with the exact output if I manage (luckily or sadly) to break the connection.

**Comment by pgueckel** [comment]
I haven't tried wyldcard's solution. I did build the module, but I'd like to back up the files that get overwritten before running make install. It is pretty confusing: there are some files in /usr/lib/firmware and some in /usr/lib/modules/<kernel version>/..., too, I think. I don't have a clue what would end up getting overwritten.

Since we are having this issue, maybe our kernels already have this driver/firmware? Like I said, the 4.9.13-20x kernels and earlier did not have the bug, only 4.9.14-200 and later.

**Comment by wyldcard** [comment]
lwfinger's driver did improve the sitiation. However there are still issues Realtek need to fix. Im basically experiencing the same issues as noted by @ferviron . Im ordering in a new wifi card and washing my hands of this.

**Comment by matchaniat** [comment]
I have the same problem with rtl8821ae, i use ArchLinux with 4.10.13-1-ARCH kernel. The connection drops after some time, the networkmanager disconnect and reconnect to the network some seconds after. Ater installing the lwfinger's driver, the situation didn't change... I can give you some logs if you want.

**Comment by Oden1** [comment]
I had problems with my rtl8821ae 1a3b:2161 that was dropping connection all the time, so often that it wasn't usable,
I have found int_clear=0 to be the parameter that makes it work.
It doesn't really matter if it's alone or in combination with other parameters.
"sudo modprobe -r rtl8821ae && sudo modprobe rtl8821ae int_clear=0"
It works for me so maybe it works for someone else as well.

**Comment by pgueckel** [comment]
I now run the 4.11.0-1.fc26.x86_64 kernel. I don't know if anything changed in the kernel modules, but things have gotten so bad that I have had to resort to restarting Network Manager and alternating with removing and loading the rtl8821ae kernel module. This barely works any longer and I am repeatedly asked for the network password, even though the Plasma Network Manager widget clearly shows that the password is stored. I read somewhere that one must restart it twice (a reason for why it fails the first time was given and why it works on the second try, but I don't recall what it was), so I have done this, having appened @Oden1's solution. Amazingly, once I get past the double restarts, this really does work! Nevertheless, there are still some disconnection problems, but they are drastically reduced.

**Comment by gyuris** [comment]
My distro is Ubuntu 17.04.
Unit is Lenovo IdeaPad 700-15
Chipset and driver is:
```
02:00.0 Network controller [0280]: Realtek Semiconductor Co., Ltd. RTL8821AE 802.11ac PCIe Wireless Network Adapter [10ec:8821]
	Subsystem: Lenovo RTL8821AE 802.11ac PCIe Wireless Network Adapter [17aa:a814]
	Kernel driver in use: rtl8821ae
```
  
The [workaround](https://bugzilla.redhat.com/show_bug.cgi?id=1435793) for me is running 4.9.12 kernel. Everything after that is wrong or worse. I tested also 4.12.2. I need to reset the driver with command mentioned above sometimes (2 or 3 times per day): 
```
sudo modprobe -r rtl8821ae && sudo modprobe rtl8821ae int_clear=0
```

**Comment by hyaochn** [comment]
[wireless-info.txt](https://github.com/lwfinger/rtlwifi_new/files/1177897/wireless-info.txt)

Same issue here.
Lenovo y520. Ubuntu 17.04
Everytime drops, I have to type
`sudo service network-manager restart`

**Comment by pgueckel** [comment]
I was reading about int_clear and it got me thinking about msi. Why had I dismissed msi from consideration? I was stuck on thinking the problem was power save [scratches head].

I have been running the computer for a few hours and the only option I have in /etc/modprobe.d/rtl8821ae.conf is:

options msi=0

msi needs to be disabled to make wifi work.

[By the way, I am now using kernel-4.12.8-300.fc26.x86_64 from koji.]

**Comment by pgueckel** [comment]
Unfortunately, my assumption has not held. I think that there might be two issues, one being the msi setting and the other being the power saving. I have tried every possible combination of the power saving settings—ips, swlps and fwlps—that I can think of (without making a rigorous study of all of the possible combinations).

I have also expanded the msi issue, if it really is a separate case, so my options have now expanded, but there is still no lasting solution:

options fwlps=0 msi=0 disable_watchdog=1 int_clear=0

What seems to work works for a little while and then the same syndrome begins anew.

**Comment by chilek** [comment]
@pgueckel did you try all of you described with lwfinger rtlwifi?

**Comment by pgueckel** [comment]
I cannot make sense of your question @chilek 
I tried  int_clear=0 fwlps=0 ips=0, but not by themselves, yet. I have given up on msi being a culprit, despite initial hopefulness. Currently, I have:

options fwlps=0 disable_watchdog=1 int_clear=0

This morning, that is working very well, but these same options last night gave me nothing but dropped connections every 1-2 minutes.

Is there somewhere where I can set the timeout so that the network doesn't disconnect right away, when the signal strength varies? I notice that the indicator (NetworkManager) sometimes shows 4 or 5 bars, typically only shows 3 bars and, just before the connection is dropped, will suddenly show 0 bars—and this while the signal strength of the wifi router has not even changed! Both my phone and blueray player indicate the highest signal strength and they connect without issue.

**Comment by djhogan** [comment]
lwfinger has an alternative driver module for realtek cards at https://github.com/lwfinger/rtlwifi_new. Basically, @chilek is wondering if you've installed this alternative driver and tried the same combination of options to see if they resolve your issue. I'm interested in this problem as well as I'm experiencing the exact same behaviour. I have an F555L Asus model.

**Comment by pepa65** [comment]
He says here he's been building from this repo here: https://github.com/lwfinger/rtlwifi_new/issues/235#issuecomment-299730609

**Comment by silgon** [comment]
@chilek answer is a nice workaround, at least for the most of the uses. However, when you download a file, the speed continues to fluctuate more that it should. Also when tested with pinging a domain, there are mini spikes of slow connectivity.

**Comment by djhogan** [comment]
@pepa65 @pgueckel sorry, I thought this was a different repo

**Comment by draskolnikova** [comment]
Sorry for jump in, I am using FC26 and stay using kernel `4.12.14-300.fc26.x86_64` for stability using `options rtl8821ae int_clear=0` on modprobe options. Have tried using latest kernel `4.13.5-200.fc26.x86_64` but it's unstable.

**Comment by christianmtr** [comment]
I just comment in [other issue](https://github.com/lwfinger/rtlwifi_new/issues/261#issuecomment-338260135) that my connection works fine on Ubuntu Mate 16.04.

**Comment by gyuris** [comment]
Today I run Ubuntu 17.10 on new 4.14 kernel. WIFI works without any problem :-).
[wireless-info.txt](https://github.com/lwfinger/rtlwifi_new/files/1466911/wireless-info.txt)

**Comment by matchaniat** [comment]
@gyuris I think it's thanks to this commit: https://github.com/torvalds/linux/commit/b8b8b16352cd90c6083033fd4487f04fae935c18

**Comment by pgueckel** [comment]
@matchaniat @lwfinger 

I resorted to buying a cable and have not used wifi all summer, as I had gotten so frustrated by the unusable connection. I upgraded to Fedora 27β when it came out about a month ago and gave wifi another try: it was worse than ever!

The new commit sounds REALLY convincing!!!

Which kernel will be the one to get it? Will it be backported to other kernels? When will it hit the fedora repo kernels?

**Comment by nikolas** [comment]
I was having this issue too, and the `b8b8b16` commit didn't resolve it for me on my Lenovo Ideapad 320-15ABR laptop. For anyone still having trouble with this, try turning off ASPM. See the thread here:

https://marc.info/?l=linux-wireless&m=150994142316118&w=2

I had to actually remove the `enable_aspm()` call in the driver to get this working properly.

**Comment by meowsus** [comment]
I too am running Ubuntu 17.10 with the newest kernel and I've noticed a bit more reliability. Oddly, though, it's only stable when I run my connection through wpa_cli.

**Comment by kladier** [comment]
Same problem for me for a while.
Are you sure it's a problem of rtl8821ae driver and not due to NetworkManager ?

**Comment by lwfinger** [comment]
That is a very good question. Using openSUSE Tumbleweed with a 4.14 kernel and NetworkManager, the device is rock solid. Performance is sometimes low, but the connection never drops.

**Comment by nikolas** [comment]
I'm using Debian with XCFE. I'm not using NetworkManager.

**Comment by lwfinger** [comment]
XFCE is a desktop, The critical user code is the network control software. What one are you using? The problem report was with wicd, and the problem went away when the user switched to NetworkManager.

**Comment by nikolas** [comment]
I have all that disabled - I'm not using NetworkManager or wicd. I tried wicd once to see if that solved my problem, but it didn't, so I uninstalled it. I'm configuring my network through debian's standard configuration files in `/etc/network/interfaces.d/`, via systemd commands like `service networking restart`, which control the interface with `ifup` and `ifdown`.

**Comment by pgueckel** [comment]
@lwfinger et al.

I loaded fedora kernel 4.14.6-300.fc27.x86_64 today. I disconnected the wired connection and tried out the wireless, albeit only for about 30 minutes so far, but I am pleasantly surprised: I have had no dropped connection occurrences! I am so optimistic that I think it has finally been resolved.

If problems resume, I will report here.

**Comment by kladier** [comment]
@nikolas which kernel version ?

At each time you upgrade your kernel, you have to rebuild rtlwifi_new and `modprobe -r rtl8821ae && modprobe rtl8821ae`

**Comment by nikolas** [comment]
@kladier I'm on kernel 4.14.7. 

Interestingly, with the same configuration on kernel 4.15.0-rc2 and 4.15.0-rc3, I am getting frequent disconnects.

**Comment by pgueckel** [comment]
Reported issue is resolved.

**Comment by silgon** [comment]
Same problem here. I know this is an old issue that has been closed. But I have a computer with ubuntu 18.04, I installed the drivers from `rtlwifi_new` to try to solve the problem but I still have it. I'll give you the output of `modinfo rtl8821ae`.

```
filename:       /lib/modules/4.15.0-88-generic/kernel/drivers/net/wireless/realtek/rtlwifi/rtl8821ae/rtl8821ae.ko                                                                                                    
firmware:       rtlwifi/rtl8821aefw_29.bin                                                                                                                                                                           
firmware:       rtlwifi/rtl8821aefw.bin 
description:    Realtek 8821ae 802.11ac PCI wireless
license:        GPL
author:         Realtek WlanFAE <wlanfae@realtek.com>
srcversion:     1F4F5BD5D1F911A750255CF
alias:          pci:v000010ECd00008821sv*sd*bc*sc*i*
alias:          pci:v000010ECd00008812sv*sd*bc*sc*i*
depends:        rtlwifi,rtl_pci,btcoexist,mac80211
retpoline:      Y
name:           rtl8821ae
vermagic:       4.15.0-88-generic SMP mod_unload
parm:           swenc:Set to 1 for software crypto (default 0)
 (bool)
parm:           ips:Set to 0 to not use link power save (default 1)
 (bool)
parm:           swlps:Set to 1 to use SW control power save (default 0)
 (bool)
parm:           fwlps:Set to 1 to use FW control power save (default 1)
 (bool)
parm:           msi:Set to 1 to use MSI interrupts mode (default 1)
 (bool)
parm:           debug:Set debug level (0-5) (default 0) (int)
parm:           disable_watchdog:Set to 1 to disable the watchdog (default 0)
 (bool)
parm:           int_clear:Set to 0 to disable interrupt clear before set (default 1)
 (bool)
```
Let me know if you know how to debug this problem. thanks!

**Comment by cryptid11** [comment]
unfortunately I have no idea, I resolve to use external dongle :/

**Comment by lwfinger** [comment]
It would be helpful if you listed the corresponding output from dmesg that shows why it is disconnecting. My 8821ae stays up for days, no matter what I throw at it.

**Comment by silgon** [comment]
Sure. After a module restart for example:
```
[80928.087091] rtlwifi: rtlwifi: wireless switch is on
[80928.089664] rtl8821ae 0000:03:00.0 wlp3s0: renamed from wlan0
[80928.117927] IPv6: ADDRCONF(NETDEV_UP): wlp3s0: link is not ready
[80928.296831] IPv6: ADDRCONF(NETDEV_UP): wlp3s0: link is not ready
[80928.376364] IPv6: ADDRCONF(NETDEV_UP): wlp3s0: link is not ready
[80931.492205] wlp3s0: authenticate with 48:53:d7:b1:93:16
[80931.492511] wlp3s0: send auth to 48:53:d7:b1:93:16 (try 1/3)
[80931.494094] wlp3s0: authenticated
[80931.495873] wlp3s0: associate with 48:53:d7:b1:93:16 (try 1/3)
[80931.499322] wlp3s0: RX AssocResp from 48:53:d7:b1:93:16 (capab=0x1411 status=0 aid=2)
[80931.499517] wlp3s0: associated
[80931.532681] IPv6: ADDRCONF(NETDEV_CHANGE): wlp3s0: link becomes ready
[80933.149937] wlp3s0: deauthenticating from 48:53:d7:b1:93:16 by local choice (Reason: 3=DEAUTH_LEAVING)
[80933.166376] wlp3s0: authenticate with 48:53:d7:b1:93:16
[80933.166679] wlp3s0: send auth to 48:53:d7:b1:93:16 (try 1/3)
[80933.168883] wlp3s0: authenticated
[80933.171896] wlp3s0: associate with 48:53:d7:b1:93:16 (try 1/3)
[80933.175378] wlp3s0: RX AssocResp from 48:53:d7:b1:93:16 (capab=0x1411 status=0 aid=2)
[80933.175573] wlp3s0: associated
[80988.312345] wlp3s0: deauthenticating from 48:53:d7:b1:93:16 by local choice (Reason: 3=DEAUTH_LEAVING)
[80988.628623] cfg80211: Loading compiled-in X.509 certificates for regulatory database
[80988.628954] cfg80211: Loaded X.509 cert 'sforshee: 00b28ddf47aef9cea7'
[80988.668507] rtl8821ae: Using firmware rtlwifi/rtl8821aefw_29.bin
[80988.668511] rtl8821ae: Using firmware rtlwifi/rtl8821aefw_wowlan.bin
[80988.669455] ieee80211 phy0: Selected rate control algorithm 'rtl_rc'
[80988.669735] rtlwifi: rtlwifi: wireless switch is on
[80988.671797] rtl8821ae 0000:03:00.0 wlp3s0: renamed from wlan0
[80988.710867] IPv6: ADDRCONF(NETDEV_UP): wlp3s0: link is not ready
[80988.890198] IPv6: ADDRCONF(NETDEV_UP): wlp3s0: link is not ready
[80988.960144] IPv6: ADDRCONF(NETDEV_UP): wlp3s0: link is not ready
[80992.067434] wlp3s0: authenticate with 48:53:d7:b1:93:16
[80992.067743] wlp3s0: send auth to 48:53:d7:b1:93:16 (try 1/3)
[80992.069399] wlp3s0: authenticated
[80992.072327] wlp3s0: associate with 48:53:d7:b1:93:16 (try 1/3)
[80992.075984] wlp3s0: RX AssocResp from 48:53:d7:b1:93:16 (capab=0x1411 status=0 aid=2)
[80992.076195] wlp3s0: associated
[80992.105231] IPv6: ADDRCONF(NETDEV_CHANGE): wlp3s0: link becomes ready
[80993.732298] wlp3s0: deauthenticating from 48:53:d7:b1:93:16 by local choice (Reason: 3=DEAUTH_LEAVING)
[80993.747997] wlp3s0: authenticate with 48:53:d7:b1:93:16
[80993.748309] wlp3s0: send auth to 48:53:d7:b1:93:16 (try 1/3)
[80993.750073] wlp3s0: authenticated
[80993.752347] wlp3s0: associate with 48:53:d7:b1:93:16 (try 1/3)
[80993.755761] wlp3s0: RX AssocResp from 48:53:d7:b1:93:16 (capab=0x1411 status=0 aid=2)
[80993.755958] wlp3s0: associated
```

**Comment by silgon** [comment]
Well. This doesn't solve the real problem but at least it solves my problem. I bought a  wifi dongle as  @cryptid11

**Comment by lwfinger** [comment]
That is one fix. Your dmesg output indicates that one end or the other is deauthenticating. That is the meaning of reason 3. Since it happens with your system, and not mine, I think that something with your AP is not compatible. What make and model are you using?

**Comment by silgon** [comment]
I was verifying my cron scripts today. Actually, the driver was not the problem. On my cron commands I had the following command `/sbin/ifconfig wlp3s0 | grep  "inet add"  || restart_rtl8821ae.sh `, it was working fine until I updated my system this weekend from ubuntu 16.04 to ubuntu 18.04 with the command `do-release-upgrade`. Thus, `inet add` is written only as `inet`, that's why I had problems, the restart script I use removes and restarts `rtl8821ae`. So, it was all due to that upgrade.

I just corrected and seems to be ok. I'll let you know if I'm mistaken. Thanks! =)
